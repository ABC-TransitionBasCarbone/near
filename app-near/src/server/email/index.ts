import { env } from "~/env";
import { type BrevoBody } from "~/types/Brevo";

const sendEmail = async (body: BrevoBody) => {
  if (env.NODE_ENV === "production") {
    return fetch("https://api.brevo.com/v3/smtp/email", {
      method: "POST",
      headers: {
        Accept: "application/json",
        "api-key": env.BREVO_API_KEY,
        "Content-Type": "application/json",
      },
      body: JSON.stringify(body),
    });
  }

  console.log(`
Email send with command:
curl --request POST \
     --url https://api.brevo.com/v3/smtp/email \
     --header 'accept: application/json' \
     --header 'api-key: <fill-with-correct-api-key>' \
     --header 'content-type: application/json' \
     --data '${JSON.stringify(body).replace(/'/g, "&apos;")}'
    `);
  return `Email template ${body.templateId} send`;
};

const EmailService = { sendEmail };

export default EmailService;
