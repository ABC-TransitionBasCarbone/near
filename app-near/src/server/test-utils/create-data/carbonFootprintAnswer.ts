import { BroadcastChannel, type CarbonFootprintAnswer } from "@prisma/client";
import { faker } from "@faker-js/faker";
import { type BuilderCarbonFootprintAnswer } from "~/types/CarbonFootprint";

export const buildCarbonFootprintAnswer = (
  surveyId: number,
  suId: number,
  data?: Partial<CarbonFootprintAnswer>,
): BuilderCarbonFootprintAnswer => ({
  broadcastChannel: faker.helpers.arrayElement(Object.values(BroadcastChannel)),
  email: Date.now() + faker.internet.email(),
  globalNote: faker.number.float(),
  transportation: faker.number.float(),
  transportationCar: faker.number.float(),
  transportationCarKm2: faker.number.float(),
  transportationCarOilConsumption100: faker.number.float(),
  transportationCarOilType: faker.number.float(),
  transportationCarElectricConsumption100: faker.number.float(),
  transportationPlane: faker.number.float(),
  transportationBicycle: faker.number.float(),
  transportationSoftMobility: faker.number.float(),
  transportationTrain: faker.number.float(),
  transportationPublicTransport: faker.number.float(),
  transportationCarLongDistance: faker.number.float(),
  transportationHollidays: faker.number.float(),
  transportationHollidaysCaravan: faker.number.float(),
  transportationHollidaysCampingCar: faker.number.float(),
  transportationHollidaysVan: faker.number.float(),
  transportationHollidaysRentalVehicle: faker.number.float(),
  transportationHollidaysCruise: faker.number.float(),
  transportationFerry: faker.number.float(),
  alimentation: faker.number.float(),
  alimentationMeals: faker.number.float(),
  alimentationLunchDinner: faker.number.float(),
  alimentationBonus: faker.number.float(),
  alimentationLocalImpact: faker.number.float(),
  alimentationLocalConsumption: faker.number.float(),
  alimentationSeasonalImpact: faker.number.float(),
  alimentationAnnualBreakfast: faker.number.float(),
  alimentationDeforestation: faker.number.float(),
  alimentationDrinks: faker.number.float(),
  alimentationHotDrinks: faker.number.float(),
  alimentationColdDrinks: faker.number.float(),
  alimentationWaste: faker.number.float(),
  alimentationWasteHabits: faker.number.float(),
  logement: faker.number.float(),
  logementConstruction: faker.number.float(),
  logementConstructionBase: faker.number.float(),
  logementConstructionCooling: faker.number.float(),
  logementConstructionRenovation: faker.number.float(),
  logementConstructionDeforestation: faker.number.float(),
  logementConstructionSoilArtificialization: faker.number.float(),
  logementElectricity: faker.number.float(),
  logementElectricityTotalConsumption: faker.number.float(),
  logementElectricitySolarAutoconsumption: faker.number.float(),
  logementElectricityCarbonIntensity: faker.number.float(),
  logementHeating: faker.number.float(),
  logementAirConditioning: faker.number.float(),
  logementSwimmingPool: faker.number.float(),
  logementOutdor: faker.number.float(),
  logementHollidays: faker.number.float(),
  logementHollidaysHotelImpact: faker.number.float(),
  logementHollidaysCampingImpact: faker.number.float(),
  logementHollidaysYouthHostelImpact: faker.number.float(),
  logementHollidaysRentalImpact: faker.number.float(),
  logementHollidaysHomeExchangeImpact: faker.number.float(),
  logementHollidaysSecondaryHomeImpact: faker.number.float(),
  divers: faker.number.float(),
  diversHouseholdAppliances: faker.number.float(),
  diversHouseholdAppliancesDevices: faker.number.float(),
  diversFurniture: faker.number.float(),
  diversFurnitureItems: faker.number.float(),
  diversFurnitureDeforestation: faker.number.float(),
  diversFurniturePreservation: faker.number.float(),
  diversDigital: faker.number.float(),
  diversDigitalInternet: faker.number.float(),
  diversDigitalDevices: faker.number.float(),
  diversTextile: faker.number.float(),
  servicesSocietal: faker.number.float(),
  servicesPublics: faker.number.float(),
  servicesMarket: faker.number.float(),
  answers: { anything: "ok" },

  airTravelFrequency: null,
  digitalIntensity: null,
  heatSource: null,
  meatFrequency: null,
  purchasingStrategy: null,
  transportationMode: null,

  distanceToBarycenter: null,
  knowSu: true,

  ...data,
  surveyId,
  suId,
});
