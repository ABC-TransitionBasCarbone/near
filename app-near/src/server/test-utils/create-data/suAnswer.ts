import {
  AgeCategory,
  AirTravelFrequency,
  BroadcastChannel,
  DigitalIntensity,
  EasyHealthAccess,
  Gender,
  HeatSource,
  MeatFrequency,
  ProfessionalCategory,
  PurchasingStrategy,
  type SuAnswer,
  TransportationMode,
} from "@prisma/client";
import { faker } from "@faker-js/faker";
import { type BuilderSuAnswer } from "~/types/SuAnswer";

export const buildSuAnswer = (
  surveyId: number,
  data: Partial<SuAnswer>,
): BuilderSuAnswer => ({
  isNeighborhoodResident: faker.datatype.boolean(),
  ageCategory: faker.helpers.arrayElement(Object.values(AgeCategory)),
  airTravelFrequency: faker.helpers.arrayElement(
    Object.values(AirTravelFrequency),
  ),
  broadcastChannel: faker.helpers.arrayElement(Object.values(BroadcastChannel)),
  digitalIntensity: faker.helpers.arrayElement(Object.values(DigitalIntensity)),
  easyHealthAccess: faker.helpers.arrayElement(Object.values(EasyHealthAccess)),
  email: faker.internet.email(),
  gender: faker.helpers.arrayElement(Object.values(Gender)),
  heatSource: faker.helpers.arrayElement(Object.values(HeatSource)),
  meatFrequency: faker.helpers.arrayElement(Object.values(MeatFrequency)),
  professionalCategory: faker.helpers.arrayElement(
    Object.values(ProfessionalCategory),
  ),
  purchasingStrategy: faker.helpers.arrayElement(
    Object.values(PurchasingStrategy),
  ),
  transportationMode: faker.helpers.arrayElement(
    Object.values(TransportationMode),
  ),
  ...data,
  surveyId,
});
