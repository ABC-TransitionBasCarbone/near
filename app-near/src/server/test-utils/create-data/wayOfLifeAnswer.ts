import {
  ActionWhenTooCold,
  ActionWhenTooHot,
  AgeCategory,
  AirTravelFrequency,
  BroadcastChannel,
  DigitalIntensity,
  EngagmentDomains,
  FoodFrequency,
  Gender,
  GiveFreeTimeToHelp,
  HeatSource,
  HomeOccupationType,
  HowManyPeopleCanIHelp,
  MeatFrequency,
  ParksUsageFrequency,
  ProfessionalCategory,
  PurchasingStrategy,
  ReasonsToEatMeat,
  ReasonsToNotBuyFrenchSeasonFood,
  ReasonsToNotChoseSecondHand,
  ReasonsUsingCar,
  TransportationMode,
  TransportMode,
  TransportModeToStation,
  TransportModeToTravel,
  TransportTime,
  TypeOfFood,
  WishesChoices,
  YesNo,
  type WayOfLifeAnswer,
} from "@prisma/client";
import { faker } from "@faker-js/faker";
import { type BuilderWayOfLifeAnswer } from "~/types/WayOfLifeAnswer";

export const buildWayOfLifeAnswer = (
  surveyId: number,
  data?: Partial<WayOfLifeAnswer>,
): BuilderWayOfLifeAnswer => ({
  accessToFoodServiceSatisfaction: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  accessToShortFoodCircuitSatisfaction: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  actionWhenTooCold: faker.helpers.arrayElements(
    Object.values(ActionWhenTooCold),
  ),
  actionWhenTooHot: faker.helpers.arrayElements(
    Object.values(ActionWhenTooHot),
  ),
  ageCategory: faker.helpers.arrayElement(Object.values(AgeCategory)),
  airTravelFrequency: faker.helpers.arrayElement(
    Object.values(AirTravelFrequency),
  ),
  associativeActivity: faker.helpers.arrayElement(Object.values(YesNo)),
  bicycleRepairShopSatisfaction: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  capacityToShareIdeaToTownHall: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  carAnPedestriansRespect: faker.helpers.arrayElement(Object.values(YesNo)),
  classicFoodMarketFrequency: faker.helpers.arrayElement(
    Object.values(FoodFrequency),
  ),
  clothesRepairShopSatisfaction: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  comment: faker.lorem.sentence(),
  confortHouseWhenHot: faker.helpers.arrayElement(Object.values(YesNo)),
  culturalActivity: faker.helpers.arrayElement(Object.values(YesNo)),
  digitalIntensity: faker.helpers.arrayElement(Object.values(DigitalIntensity)),
  easyBicycle: faker.helpers.arrayElement(Object.values(YesNo)),
  easyPublicTransports: faker.helpers.arrayElement(Object.values(YesNo)),
  easyToLeaveCityWithTranports: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  easyWalking: faker.helpers.arrayElement(Object.values(YesNo)),
  electronicRepairShopSatisfaction: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  foodMarketZone: faker.lorem.word(),
  gender: faker.helpers.arrayElement(Object.values(Gender)),
  giveFreeTimeToHelp: faker.helpers.arrayElement(
    Object.values(GiveFreeTimeToHelp),
  ),
  groceryStoreFrequency: faker.helpers.arrayElement(
    Object.values(FoodFrequency),
  ),
  hardDiscountFrequency: faker.helpers.arrayElement(
    Object.values(FoodFrequency),
  ),
  heatSource: faker.helpers.arrayElement(Object.values(HeatSource)),
  hobbiesSpaces: faker.helpers.arrayElement(Object.values(YesNo)),
  hobbyZone: faker.lorem.word(),
  homeDeliveryFoodStoreFrequency: faker.helpers.arrayElement(
    Object.values(FoodFrequency),
  ),
  homeOccupationType: faker.helpers.arrayElement(
    Object.values(HomeOccupationType),
  ),
  howManyPeopleCanIHelp: faker.helpers.arrayElement(
    Object.values(HowManyPeopleCanIHelp),
  ),
  hypermarketFrequency: faker.helpers.arrayElement(
    Object.values(FoodFrequency),
  ),
  ideaEasyTalk: faker.helpers.arrayElement(Object.values(YesNo)),
  knowSu: faker.helpers.arrayElement([true, false]),
  localShopsToMeetYourNeeds: faker.helpers.arrayElement(Object.values(YesNo)),
  meatFrequency: faker.helpers.arrayElement(Object.values(MeatFrequency)),
  neighborhoodLife: faker.helpers.arrayElement(Object.values(YesNo)),
  neighborhoodOrganicMarketSatisfaction: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  neighborhoodOrganicProductsSatisfaction: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  neighborhoodSeasonFruitAndVegetablesSatisfaction: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  noInformationOnCitizenParticipation: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  notColdHouse: faker.helpers.arrayElement(Object.values(YesNo)),
  notTooMuchTraffic: faker.helpers.arrayElement(Object.values(YesNo)),
  organicOrSolidarityFoodStoreFrequency: faker.helpers.arrayElement(
    Object.values(FoodFrequency),
  ),
  otherFoodFrequencyInformation: faker.lorem.sentence(),
  otherFoodSatisfactionInformation: faker.lorem.sentence(),
  otherHousingInformation: faker.lorem.sentence(),
  otherLocalPoliticInformation: faker.lorem.sentence(),
  otherMutualAidInformation: faker.lorem.sentence(),
  otherNeighborhoodLifeInformation: faker.lorem.sentence(),
  otherParksInformation: faker.lorem.sentence(),
  otherRepairShopSatisfactionInformation: faker.lorem.sentence(),
  otherServicesInformation: faker.lorem.sentence(),
  otherTransportationInformation: faker.lorem.sentence(),
  parksUsageFrequency: faker.helpers.arrayElement(
    Object.values(ParksUsageFrequency),
  ),
  preferBuyFrenchAndSeasonFood: faker.helpers.arrayElement(
    Object.values(WishesChoices),
  ),
  preferSecondHand: faker.helpers.arrayElement(Object.values(WishesChoices)),
  privateOrShareFieldToFarm: faker.helpers.arrayElement(Object.values(YesNo)),
  professionalCategory: faker.helpers.arrayElement(
    Object.values(ProfessionalCategory),
  ),
  publicServicesPresence: faker.helpers.arrayElement(Object.values(YesNo)),
  purchasingStrategy: faker.helpers.arrayElement(
    Object.values(PurchasingStrategy),
  ),
  reasonsToContinueUsingCar: faker.helpers.arrayElements(
    Object.values(ReasonsUsingCar),
  ),
  reasonsToEatMeat: faker.helpers.arrayElements(
    Object.values(ReasonsToEatMeat),
  ),
  reasonsToNotBuyFrenchAndSeasonFood: faker.helpers.arrayElements(
    Object.values(ReasonsToNotBuyFrenchSeasonFood),
  ),
  reasonsToNotChoseSecondHand: faker.helpers.arrayElements(
    Object.values(ReasonsToNotChoseSecondHand),
  ),
  remoteWorkingWeeklyFrequency: faker.helpers.arrayElement([
    "1",
    "2",
    "3",
    "4",
    "5",
  ]),
  secondHandShopSatisfaction: faker.helpers.arrayElement(Object.values(YesNo)),
  servicesToShareOrRentObjects: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  specializedFoodStoreFrequency: faker.helpers.arrayElement(
    Object.values(FoodFrequency),
  ),
  supermarketFrequency: faker.helpers.arrayElement(
    Object.values(FoodFrequency),
  ),
  transportationMode: faker.helpers.arrayElement(
    Object.values(TransportationMode),
  ),
  transportModeToBuyFood: faker.helpers.arrayElement(
    Object.values(TransportMode),
  ),
  transportModeToGoToStation: faker.helpers.arrayElement(
    Object.values(TransportModeToStation),
  ),
  transportModeToHobby: faker.helpers.arrayElement(
    Object.values(TransportMode),
  ),
  transportModeToTravel: faker.helpers.arrayElement(
    Object.values(TransportModeToTravel),
  ),
  transportModeToWork: faker.helpers.arrayElement(Object.values(TransportMode)),
  transportTimeToBuyFood: faker.helpers.arrayElement(
    Object.values(TransportTime),
  ),
  transportTimeToHobby: faker.helpers.arrayElement(
    Object.values(TransportTime),
  ),
  transportTimeToWork: faker.helpers.arrayElement(Object.values(TransportTime)),
  typeOfFood: faker.helpers.arrayElement(Object.values(TypeOfFood)),
  vegetalParksSatisfaction: faker.helpers.arrayElement(Object.values(YesNo)),
  wantToParticipateToCivicInitiatives: faker.helpers.arrayElement(
    Object.values(YesNo),
  ),
  wantToReduceCarUsage: faker.helpers.arrayElement(
    Object.values(WishesChoices),
  ),
  wantToReduceMeatAndFish: faker.helpers.arrayElement(
    Object.values(WishesChoices),
  ),
  workZone: faker.lorem.word(),
  yourEngagmentDomains: faker.helpers.arrayElements(
    Object.values(EngagmentDomains),
  ),

  emailApiCalled: false,
  broadcastChannel: faker.helpers.arrayElement(Object.values(BroadcastChannel)),
  email: Date.now() + faker.internet.email(),
  ...data,
  surveyId,
});
