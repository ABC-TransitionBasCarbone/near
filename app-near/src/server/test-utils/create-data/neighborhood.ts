import { SurveyPhase } from "@prisma/client";
import { db } from "~/server/db";

export const createNeighborhood = async (surveyName: string) => {
  const survey = await db.survey.create({
    data: {
      name: surveyName,
      phase: SurveyPhase.STEP_2_SU_SURVERY,
      sampleTarget: 3,
    },
  });

  await db.$queryRawUnsafe(`CREATE OR REPLACE VIEW quartiers AS
      SELECT 
        ${survey.id} AS survey_id,
        ARRAY['75014', '75015', '75016']::text[] AS iris_selectors,
        1100::double precision AS population_sum_with_under_fifteen,
        1000::double precision AS population_sum,
        180::double precision AS p21_pop1529_sum,
        230::double precision AS p21_pop3044_sum,
        170::double precision AS p21_pop4559_sum,
        200::double precision AS p21_pop6074_sum,
        220::double precision AS p21_pop75p_sum,
        220::double precision AS c21_pop15p_cs1_sum,
        110::double precision AS c21_pop15p_cs2_sum,
        180::double precision AS c21_pop15p_cs3_sum,
        150::double precision AS c21_pop15p_cs4_sum,
        50::double precision AS c21_pop15p_cs5_sum,
        120::double precision AS c21_pop15p_cs6_sum,
        80::double precision AS c21_pop15p_cs7_sum,
        90::double precision AS c21_pop15p_cs8_sum,
        560::double precision AS population_femme_sum,
        440::double precision AS population_homme_sum,
        0::double precision AS population_sum_threshold_3p,
        0::double precision AS population_sum_threshold_4p,
        0::double precision AS population_sum_threshold_5p,
        0::double precision AS population_sum_threshold_4_5p
        ;`);

  return survey;
};
