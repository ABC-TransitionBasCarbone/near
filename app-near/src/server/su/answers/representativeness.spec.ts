import {
  AgeCategory,
  AirTravelFrequency,
  DigitalIntensity,
  EasyHealthAccess,
  Gender,
  HeatSource,
  MeatFrequency,
  ProfessionalCategory,
  PurchasingStrategy,
  type Survey,
  TransportationMode,
} from "@prisma/client";
import { db } from "~/server/db";
import representativenessService from "./representativeness";
import { TRPCError } from "@trpc/server";
import { createNeighborhood } from "~/server/test-utils/create-data/neighborhood";

describe("representativeness", () => {
  let survey: Survey;
  const surveyName = "survey-test-representativeness";

  beforeEach(async () => {
    await db.suAnswer.deleteMany().catch(() => null);
    await db.suData.deleteMany().catch(() => null);
    await db.survey.delete({ where: { name: surveyName } }).catch(() => null);

    survey = await createNeighborhood(surveyName);
  });

  it("should return null when no answer is available", async () => {
    const result = await representativenessService.representativeness(
      survey.id,
    );

    expect(result).toBe(null);
  });

  it("should throw error when surveyId is not found", async () => {
    expect.assertions(2);
    try {
      await representativenessService.representativeness(7895);
    } catch (error) {
      if (error instanceof TRPCError) {
        expect(error.code).toBe("NOT_FOUND");
        expect(error.message).toBe("survey 7895 not found");
      }
    }
  });

  it("should return representativeness when answers quantity is below sampleTarget", async () => {
    await db.suAnswer.createMany({
      data: [
        {
          ageCategory: AgeCategory.ABOVE_75,
          airTravelFrequency: AirTravelFrequency.ABOVE_3,
          broadcastChannel: "mail_campaign",
          digitalIntensity: DigitalIntensity.INTENSE,
          easyHealthAccess: EasyHealthAccess.EASY,
          gender: Gender.MAN,
          heatSource: HeatSource.ELECTRICITY,
          meatFrequency: MeatFrequency.MAJOR,
          professionalCategory: ProfessionalCategory.CS1,
          purchasingStrategy: PurchasingStrategy.MIXED,
          transportationMode: TransportationMode.CAR,
          surveyId: survey.id,
        },
        {
          ageCategory: AgeCategory.ABOVE_75,
          airTravelFrequency: AirTravelFrequency.ABOVE_3,
          broadcastChannel: "mail_campaign",
          digitalIntensity: DigitalIntensity.INTENSE,
          easyHealthAccess: EasyHealthAccess.EASY,
          gender: Gender.OTHER,
          heatSource: HeatSource.ELECTRICITY,
          meatFrequency: MeatFrequency.MAJOR,
          professionalCategory: ProfessionalCategory.CS1,
          purchasingStrategy: PurchasingStrategy.MIXED,
          transportationMode: TransportationMode.CAR,
          surveyId: survey.id,
        },
      ],
    });

    const result = await representativenessService.representativeness(
      survey.id,
    );

    expect(result).toMatchSnapshot();
  });

  it("should return representativeness when answers quantity is above sampleTarget", async () => {
    await db.suAnswer.createMany({
      data: [
        {
          ageCategory: AgeCategory.ABOVE_75,
          airTravelFrequency: AirTravelFrequency.ABOVE_3,
          broadcastChannel: "mail_campaign",
          digitalIntensity: DigitalIntensity.INTENSE,
          easyHealthAccess: EasyHealthAccess.EASY,
          gender: Gender.MAN,
          heatSource: HeatSource.ELECTRICITY,
          meatFrequency: MeatFrequency.MAJOR,
          professionalCategory: ProfessionalCategory.CS1,
          purchasingStrategy: PurchasingStrategy.MIXED,
          transportationMode: TransportationMode.CAR,
          surveyId: survey.id,
        },
        {
          ageCategory: AgeCategory.ABOVE_75,
          airTravelFrequency: AirTravelFrequency.ABOVE_3,
          broadcastChannel: "mail_campaign",
          digitalIntensity: DigitalIntensity.INTENSE,
          easyHealthAccess: EasyHealthAccess.EASY,
          gender: Gender.MAN,
          heatSource: HeatSource.ELECTRICITY,
          meatFrequency: MeatFrequency.MAJOR,
          professionalCategory: ProfessionalCategory.CS1,
          purchasingStrategy: PurchasingStrategy.MIXED,
          transportationMode: TransportationMode.CAR,
          surveyId: survey.id,
        },
        {
          ageCategory: AgeCategory.ABOVE_75,
          airTravelFrequency: AirTravelFrequency.ABOVE_3,
          broadcastChannel: "mail_campaign",
          digitalIntensity: DigitalIntensity.INTENSE,
          easyHealthAccess: EasyHealthAccess.EASY,
          gender: Gender.MAN,
          heatSource: HeatSource.ELECTRICITY,
          meatFrequency: MeatFrequency.MAJOR,
          professionalCategory: ProfessionalCategory.CS1,
          purchasingStrategy: PurchasingStrategy.MIXED,
          transportationMode: TransportationMode.CAR,
          surveyId: survey.id,
        },
        {
          ageCategory: AgeCategory.ABOVE_75,
          airTravelFrequency: AirTravelFrequency.ABOVE_3,
          broadcastChannel: "mail_campaign",
          digitalIntensity: DigitalIntensity.INTENSE,
          easyHealthAccess: EasyHealthAccess.EASY,
          gender: Gender.OTHER,
          heatSource: HeatSource.ELECTRICITY,
          meatFrequency: MeatFrequency.MAJOR,
          professionalCategory: ProfessionalCategory.CS1,
          purchasingStrategy: PurchasingStrategy.MIXED,
          transportationMode: TransportationMode.CAR,
          surveyId: survey.id,
        },
      ],
    });

    const result = await representativenessService.representativeness(
      survey.id,
    );

    expect(result).toMatchSnapshot();
  });
});
