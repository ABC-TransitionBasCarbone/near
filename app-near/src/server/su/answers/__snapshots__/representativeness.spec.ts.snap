// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`representativeness should return representativeness when answers quantity is above sampleTarget 1`] = `
{
  "above_75": -78,
  "cs1": -78,
  "cs2": 11,
  "cs3": 18,
  "cs4": 15,
  "cs5": 5,
  "cs6": 12,
  "cs7": 8,
  "cs8": 9,
  "from_15_to_29": 18,
  "from_30_to_44": 23,
  "from_45_to_59": 17,
  "from_60_to_74": 20,
  "man": -56.00000000000001,
  "woman": 56.00000000000001,
}
`;

exports[`representativeness should return representativeness when answers quantity is below sampleTarget 1`] = `
{
  "above_75": -28.000000000000004,
  "cs1": -28.000000000000004,
  "cs2": 11,
  "cs3": 18,
  "cs4": 15,
  "cs5": 5,
  "cs6": 12,
  "cs7": 8,
  "cs8": 9,
  "from_15_to_29": 18,
  "from_30_to_44": 23,
  "from_45_to_59": 17,
  "from_60_to_74": 20,
  "man": -6,
  "woman": 56.00000000000001,
}
`;
